server/routes/index.ts

```diff
@@ -1,13 +1,23 @@
 import type { RequestHandler, Router } from 'express'
 
 import asyncMiddleware from '../middleware/asyncMiddleware'
+import CommunityApiService from '../services/communityApiService'
+import IntegrationSamplesRoutes from './integrationSamples'
 
-export default function routes(router: Router): Router {
+interface RouteProvider {
+  [key: string]: RequestHandler
+}
+
+export default function routes(router: Router, communityApiService: CommunityApiService): Router {
   const get = (path: string, handler: RequestHandler) => router.get(path, asyncMiddleware(handler))
 
+  const integrationSamples: RouteProvider = IntegrationSamplesRoutes(communityApiService)
```

Comment:
I'm wondering what the `RouteProvider` interface gives us here - it seems to me like we're throwing away some useful type information by downcasting the `IntegrationSamplesRoutes` to a `RouteProvider` - namely that it has a `viewDeliusUserSample` property. But I'm still very much getting to grips with TS and possibly applying faulty mappings from type systems in other languages…

Reply:
my thinking here was just, "let's see if we can create a type for these 'controller' objects that will encourage consistency". i wouldn't say we're downcasting, just saying that `IntegrationSamplesRoutes` implements `RouteProvider`